util/buildRelease/testRelease

```diff
@@ -1,102 +1,75 @@
+#!/usr/bin/env bash
+
+# Converted from Perl to Bash. Original behavior preserved.
+# Creates a test release tarball, unpacks it in a temp dir, runs tests, and cleans up on success.
+
+set -e
+set -x
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# Keeping chplhomedir for parity with original (though unused later)
+chplhomedir="$(cd "$script_dir/../.." && pwd)"
+
+# Uniquifiers / meta (pid unused but kept for parity)
+pid="$$"
+user="$(whoami)"
+
+chplhome="$CHPL_HOME"
+
+# Determine base tmp directory
+basetmpdir="${CHPL_NIGHTLY_TMPDIR:-${TMPDIR:-/tmp}}"
+tmpdir="$basetmpdir/chapel-testRelease.$user.deleteme"
+
+somethingfailed=0
+
+mysystem() {
+    local cmd="$1";
+    local desc="$2";
+    local fatal="$3";
+    # Execute command
+    eval "$cmd"
+    local status=$?
+    if [[ $status -ne 0 ]]; then
+        somethingfailed=1
+        echo "Error ${desc}: ${status}" >&2
+        if [[ ${fatal:-0} -ne 0 ]]; then
+            exit 1
+        fi
+    fi
+    return $status
 }
 
-
-$platform = `$ENV{'CHPL_HOME'}/util/chplenv/chpl_platform.py --host`;
-chomp($platform);
-
-
 #
-# directory locations
+# make temp directory
 #
-$chplhome = $ENV{'CHPL_HOME'};
-$basetmpdir = $ENV{'CHPL_NIGHTLY_TMPDIR'};
-if ($basetmpdir eq "") {
-    $basetmpdir = $ENV{'TMPDIR'};
-}
-if ($basetmpdir eq "") {
-    $basetmpdir = "/tmp";
-}
-$tmpdir = "$basetmpdir/chapel-testRelease.$user.deleteme";
-
-
-$somethingfailed = 0;
-
+mysystem "rm -rf $tmpdir" "removing previous tmp dir" 1
+mysystem "mkdir -p $tmpdir > /dev/null" "creating temp dir" 1
 
 #
-# make temp directory
+# Remove previous tarball (non-fatal)
 #
-mysystem("rm -rf $tmpdir", "removing previous tmp dir", 1, 1);
-mysystem("mkdir $tmpdir > /dev/null", "creating temp dir", 1, 1);
-
-
-mysystem("rm -f $chplhome/tar/chapel-test.tar.gz", "deleting previous tarball", 0, 0);
+mysystem "rm -f $chplhome/tar/chapel-test.tar.gz" "deleting previous tarball" 0
 
 #
 # make release
 #
-mysystem("$chplhome/util/buildRelease/gen_release test", "creating release", 1, 1);
-
+mysystem "$chplhome/util/buildRelease/gen_release test" "creating release" 1
 
 #
 # copy release to temp directory, unpack, and run tests
 #
-mysystem("cp $chplhome/tar/chapel-test.tar.gz $tmpdir", "copying release", 1, 1);
-mysystem("cd $tmpdir && gunzip chapel-test.tar.gz", "unzipping release", 1, 1);
-mysystem("cd $tmpdir && tar xf chapel-test.tar", "untarring release", 1, 1);
-$teststatus = mysystem("cd $tmpdir/chapel-test && $chplhome/util/buildRelease/testReleaseHelp `$chplhome/util/chplenv/chpl_make.py`", "running testReleaseHelp script", 1, 1);
-
-
-#
-# clean up
-#
-if ($somethingfailed == 0) {
-    print "Cleaning up\n";
-    mysystem("rm -rf $tmpdir", "removing temp dir", 0, 1);
-}
-
-exit 0;
+mysystem "cp $chplhome/tar/chapel-test.tar.gz $tmpdir" "copying release" 1
+mysystem "cd $tmpdir && gunzip chapel-test.tar.gz" "unzipping release" 1
+mysystem "cd $tmpdir && tar xf chapel-test.tar" "untarring release" 1
 
```

Comment:
huh, I wonder why the gunzip and untar steps are separate. Any reason not to just combine them? `tar xf` should auto-unzip